### PR TITLE
chore: Ensures that examples compile

### PR DESCRIPTION
**Pull request by lantoli**
## Proposed changes

Ensures that TypeScript examples compile.   

Also the examples are fixed in this PR.

_Jira ticket:_ CLOUDP-251764

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] I have tested the CDK constructor in a CFN stack. See [TESTING.md](../TESTING.md)
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

## Further comments

Example of fail logs:
log

### CODE REVIEW

File: examples/l3-resources/encryption-at-rest-express.ts [hunk 1]
```diff
@@ -22,35 +22,35 @@ export class CdkTestingStack extends cdk.Stack {
     const atlasProps = this.getContextProps();
     const atlasBasicPrivateEndpointasBasic = AtlasEncryptionAtRestExpress(this, 'AtlasBasic', {
       cluster: {
-          name: atlasProps.clusterName,,
```
Comment (lantoli) [inline examples/l3-resources/encryption-at-rest-express.ts:25]: it had two consecutive commas

File: scripts/check-examples.sh
Comment (oarbusi) [inline scripts/check-examples.sh:1]: Nice!

File: src/l3-resources/atlas-basic-private-endpoint/index.ts [hunk 2]
```diff
@@ -61,12 +61,14 @@ export class AtlasBasicPrivateEndpoint extends Construct {
...
 
+    const region = props.region || privateEndpointDefaults.region;
+
     this.privateEndpointService = new CfnPrivateEndpointService(
       this,
       "atlas-private-endpoint-service-".concat(id),
       {
         projectId: this.atlasBasic.mProject.attrId,
-        region: props.region.toUpperCase().replace(/-/g, "_"),
+        region: region.toUpperCase().replace(/-/g, "_"),
```
Comment (oarbusi) [inline src/l3-resources/atlas-basic-private-endpoint/index.ts:71]: was this wrong before? I guess you found the issue while compiling examples?

**Review by oarbusi — APPROVED** [review]
LGTM

File: src/l3-resources/atlas-basic-private-endpoint/index.ts (hunk 2 above)
Comment (lantoli) [reply src/l3-resources/atlas-basic-private-endpoint/index.ts:71]: yes, I found it in that way. The thing is that we said it was optional but it really wasn't, so I've changed the code to allow region to be optional

region info:
```
  /**
   * @description AWS Region
   * @type {string}
   * @default us-east-1
   * @memberof AtlasPrivateEndpointProps
   */
  readonly region?: string;

```

Comment (lantoli) [reply src/l3-resources/atlas-basic-private-endpoint/index.ts:71]: the key to make it optional is here:
```
const region = props.region || privateEndpointDefaults.region;
```

**Pull request merged**